docs/network/network.md

```diff
@@ -1,27 +1,47 @@
+## Overview of the Miden network
 
-## Miden Nodes
-Operators will run Miden Nodes. Operators ensure integrity of the Account, Note and Nullifier State - which represent the state of Polygon Miden. Operators can execute and proof transactions against single accounts and they can verify proofs of locally executed transactions. Furthermore, the operator compresses the proofs in several steps up to a single proofs that gets published and verified on the Verifier contract. Operators also watch events emitted by the Bridge Contract to detect deposits and withdrawals.
+![Miden architecture overview](../img/network/architecture-overview.svg)
 
-To manage all of this, Miden Nodes have different modules. The Node orchestrates a Tx Prover, a Tx Aggregator and a Block Producer. The Tx Prover executes and proves transactions, like in the Miden Client. The Tx Aggregator can batch multiple proofs together to reduce the final state proof size using recursive proving. The Block Producer exposes the RPC interface to the user. The Block Producer collects transactions in the Tx Pool and stores the state of Polygon Miden in its three databases (Accounts, Notes, Nullifiers).
+## Miden clients
+
+Users run Miden clients and they provide an interface for wallets representing accounts on Miden. 
+
+Miden clients can execute and prove transactions with the tx prover. They can handle arbitrary signature schemes. The default is [Falcon](https://falcon-sign.info/). There is a wallet user interface, a database that stores account data locally, and the required smart contract code that represents the account on Miden.
+
+## Miden nodes
+
+Operators run Miden Nodes. 
+
+Operators ensure integrity of account, note, and nullifier states - all of which represent the state of Polygon Miden. Operators can execute and prove transactions against single accounts and they can also verify proofs of locally executed transactions. 
+
+Furthermore, the operator compresses the proofs in several steps up to a single proof that gets published and verified on the verifier contract. Operators also watch events emitted by the bridge contract to detect deposits and withdrawals.
+
+### Node modules
+
+To manage all of this, Miden nodes have separate modules. 
+
+- Tx prover: Executes and proves transactions, like the Miden client.
+- Tx aggregator: Batches multiple proofs together to reduce the final state proof size using recursive proving.
+- Block producer: exposes the RPC interface to the user and collects transactions in the tx pool and stores the state of Polygon Miden in its three databases (accounts, notes, and nullifiers).
+
+## Verifier contract
```

Comment:
You can not know that. But can you add here 

`Note: Polygon Miden will be integrated into the Polygon Aggregation Layer. The specific design is not finalized yet.`